### ISSUE TITLE
Error with URLs

### ISSUE THREAD

**Issue by debugpoint136** [GMOD/tabix-js#22]
Hi, 

Pardon my naive question, but is there any specific config settings to use with data available through URL?  

I am using the example code from your README, it works fine if the data is in the same local folder, but errors out if available through some URL. What am I missing?

**Error** - `dpuru@dpuru  ~/scratch/tmp/gmod-tabix-test  node gmod-tabix.js 
(node:19975) UnhandledPromiseRejectionWarning: Error: ENOENT: no such file or directory, open 'http://target.wustl.edu/ENCFF000ARO_ALL.iteres.loci.gz'
(node:19975) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:19975) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
(node:19975) UnhandledPromiseRejectionWarning: Error: ENOENT: no such file or directory, open 'http://target.wustl.edu/ENCFF000ARO_ALL.iteres.loci.gz.tbi'
(node:19975) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 2)
{ Error: ENOENT: no such file or directory, open 'http://target.wustl.edu/ENCFF000ARO_ALL.iteres.loci.gz.tbi'
  errno: -2,
  code: 'ENOENT',
  syscall: 'open',
  path: 'http://target.wustl.edu/ENCFF000ARO_ALL.iteres.loci.gz.tbi' }`

I confirmed that .tbi file is indeed present on the URL.

Here is the code I am trying - 
`const {TabixIndexedFile} = require('@gmod/tabix')

const tbiIndexed = new TabixIndexedFile({ 
    path: 'http://target.wustl.edu/ENCFF000ARO_ALL.iteres.loci.gz',
    tbiPath: 'http://target.wustl.edu/ENCFF000ARO_ALL.iteres.loci.gz.tbi' 
})

async function go() {
const lines = []
try {
    await tbiIndexed.getLines('MER41B',0,1, (line, fileOffset) => lines.push(line))
    console.log(lines);
} catch (error) {
    console.log(error)
}
}

go();`

I am very excited to use this module to access bedgraph files from my browser.

Thanks in advance!!

**Comment by cmdcolin** [comment]
Instead of path (which only works for local files), you will want to use the "filehandle and tbiFilehandle" arguments to the constructor

You can install generic-filehandle from https://github.com/GMOD/generic-filehandle and say something like

```
import {RemoteFile} from 'generic-filehandle'
const tbiIndexed = new TabixIndexedFile({
    filehandle: new RemoteFile('http://target.wustl.edu/ENCFF000ARO_ALL.iteres.loci.gz'),
    tbiFilehandle: new RemoteFile('http://target.wustl.edu/ENCFF000ARO_ALL.iteres.loci.gz.tbi')
})
```

We will try and improve some documentation on our end for this

**Comment by debugpoint136** [comment]
Thanks for prompt response Colin!

I tried what you suggested above. I am now seeing a different error - 

`remoteFile.js:324 Uncaught (in promise) Error: unable to determine size of file at http://target.wustl.edu/ENCFF000ARO_ALL.iteres.loci.gz
    at RemoteFile._callee4$ (remoteFile.js:324)
    at tryCatch (runtime.js:45)
    at Generator.invoke [as _invoke] (runtime.js:271)
    at Generator.prototype.<computed> [as next] (runtime.js:97)
    at asyncGeneratorStep (asyncToGenerator.js:3)
    at _next (asyncToGenerator.js:25)`

(I am trying this from a react component in browser) [code here](https://gist.github.com/debugpoint136/ac1570d6abdb7d5dc98a91bd958ee2e8)

Interestingly, if I comment everything in componentDidMount and only leave the below line uncommented, that works and returns a valid number.
`const numLines = await tbiIndexed.lineCount('MER41B');`

**Comment by cmdcolin** [comment]
I can confirm, it might have to do with some CORS settings but i'll see if we can check it out (ideally wouldn't be dependent on CORS settings)

**Comment by debugpoint136** [comment]
I suspect CORS settings as well. To test it, I copied the same file and .tbi file to an S3 bucket and enabled CORS. I have updated the above mentioned [gist](https://gist.github.com/debugpoint136/ac1570d6abdb7d5dc98a91bd958ee2e8) for your ease of testing. 
I noted that this did not remove the error. Highly appreciate your time. Thanks!

**Comment by garrettjstevens** [comment]
The `Error: unable to determine size of file` is because it's not able to read the `content-range` header from the file request. I made a copy of these files into the JBrowse S3 bucket and the error went away, so it looks like there still might be a CORS configuration problem. A CORS configuration for S3 that generally works can be found here: https://jbrowse.org/docs/cors.html#amazon-s3

Also, I had to change the range to start at 0 instead of 1 to be able to retrieve the lines.
```diff
- await tbiIndexed.getLines('MER41B', 1, 20, (line, fileOffset) => lines.push(line));
+ await tbiIndexed.getLines('MER41B', 0, 20, (line, fileOffset) => lines.push(line));
```

This also seems to have uncovered a bug where `getHeader()` returns the first line of the file if there is no header. We'll have to look into that.
